### ISSUE TITLE
proposal: use ~ instead of =~ for regular expression constraints

### ISSUE THREAD

**Issue by myitcv** [cue-lang/cue#2001]
Currently (per [the spec](https://tip.cuelang.org/docs/references/spec/#comparison-operators)), the `=~` operator is defined as follows:

```
s =~ r is true if s matches the regular expression r.
```

This issue tracks changing that operator to instead be simply `~`, with the same definition. 

### Motivation etc.

(to follow from @mpvl)

**Comment by seh** [comment]
What is the motivation for the change? I find that `=~` goes along with other inequality operators such as `<=` and `>=`, though perhaps that analogy would argue in favor of `~=`.

**Comment by myitcv** [comment]
> What is the motivation for the change?

Thanks. I'm just in the process of creating the umbrella issues we referenced from [Community Call #1](https://github.com/cue-lang/cue/discussions/1958), hence why some issues are a bit light on detail. I'll defer to @mpvl to flesh this out (and will add a section above to highlight that it's pending).

**Comment by mpvl** [comment]
@seh, one big motivation is how ugly it is to use regexps in a pattern constraint with an alias:
```
[X= =~"regexp"]: T
```

**Comment by seh** [comment]
I agree: That sure _is_ ugly. I assume that you don’t like the transposed proposal any better:
```cue
[X= ~="regexp"]: T
```

**Comment by eonpatapon** [comment]
I agree it would be better in the context of pattern constraint.

I also regularly forgot if it is `=~` or `~=` when writing configs.

Logically I think `=~` is the correct way as we also have `!~`. Would be awkward to have `~=` and `!~`.

Having `~` and `!~` feels ok to me.
